lib/content/mixin.js

```diff
@@ -130,8 +130,28 @@ export default {
       type: Boolean,
       default: false,
     },
+    timeSpent: {
+      type: Number,
+      default: 0,
+    },
+    duration: {
+      type: Number,
+      default: null,
+    },
   },
   computed: {
+    // For when we want to force a renderer to use time-based progress (e.g. instead of % completed)
+    forceTimeBasedProgress() {
```

Comment:
A thought that occurred to me while reviewing your other PR - maybe we should make `forceDurationBasedProgress` here, so that the names match?